src/mercury_prof_interface.c

```diff
@@ -0,0 +1,281 @@
+  *bind = (*val).pvar_bind;
+  *continuous = (*val).continuous;
+
+  return HG_SUCCESS;
+}
+
+/*---------------------------------------------------------------------------*/
+hg_return_t 
+HG_Prof_pvar_session_create(hg_class_t *hg_class, hg_prof_pvar_session_t *session) {
+
+  struct hg_private_class *hg_private_class = (struct hg_private_class *)(hg_class);
+
+  if(!hg_prof_get_is_initialized(hg_private_class))
+    return HG_NA_ERROR;
+
+  (*session) = (hg_prof_pvar_session_t)malloc(sizeof(struct hg_prof_pvar_session));
+  (*session)->pvar_handle_array = (hg_prof_pvar_handle_t *)malloc(sizeof(hg_prof_pvar_handle_t)*NUM_PVARS);
+  hg_private_class->session = (*session);
+
+  hg_prof_set_session_is_initialized((*session));
+
+  return HG_SUCCESS;
+}
+
+/*---------------------------------------------------------------------------*/
+hg_return_t 
+HG_Prof_pvar_session_destroy(hg_class_t *hg_class, hg_prof_pvar_session_t *session) {
+
+  struct hg_private_class *hg_private_class = (struct hg_private_class *)(hg_class);
+
+  if(!hg_prof_get_is_initialized(hg_private_class))
+    return HG_NA_ERROR;
+
+  free((*session)->pvar_handle_array);
+  free((*session));
+  hg_private_class->session = NULL;
+
+  return HG_SUCCESS;
+}
+/*---------------------------------------------------------------------------*/
+hg_return_t 
+HG_Prof_pvar_handle_alloc(hg_prof_pvar_session_t session, int pvar_index, void *obj_handle, hg_prof_pvar_handle_t *handle, int *count) {
+
+  if(!hg_prof_get_session_is_initialized(session))
+    return HG_NA_ERROR;
+
+  /* Only supporting a null bind type at the moment */
+  assert(obj_handle == NULL);
+
+  struct hg_prof_pvar_session s = *session;
+  unsigned int key = pvar_index;
+  hg_prof_pvar_data_t * val;
+
+  s.pvar_handle_array[pvar_index] = (hg_prof_pvar_handle_t)malloc(sizeof(struct hg_prof_pvar_handle));
+  val = hg_prof_pvar_table_lookup(key);
+
+  /* Copy out information from the internal PVAR hash table */
+  (*s.pvar_handle_array[pvar_index]).pvar_class = (*val).pvar_class;
+  (*s.pvar_handle_array[pvar_index]).pvar_datatype = (*val).pvar_datatype;
+  (*s.pvar_handle_array[pvar_index]).pvar_bind = (*val).pvar_bind;
+  (*s.pvar_handle_array[pvar_index]).continuous = (*val).continuous;
+  (*s.pvar_handle_array[pvar_index]).is_started = 0;
+  (*s.pvar_handle_array[pvar_index]).addr = (*val).addr;
+  if((*val).continuous)
+    (*s.pvar_handle_array[pvar_index]).is_started = 1;
+  strcpy((*s.pvar_handle_array[pvar_index]).name, (*val).name);
+  strcpy((*s.pvar_handle_array[pvar_index]).description, (*val).description);
+  *count = (*val).count;
+
+  /* Return handle */
+  *handle = s.pvar_handle_array[pvar_index];
+
+  return HG_SUCCESS;
+}
+
+/*---------------------------------------------------------------------------*/
+hg_return_t 
+HG_Prof_pvar_handle_free(hg_prof_pvar_session_t session, int pvar_index, hg_prof_pvar_handle_t *handle) {
+
+  if(!hg_prof_get_session_is_initialized(session))
+    return HG_NA_ERROR;
```

Comment:
that seems like the wrong error code :) HG_NA_ERROR means it's an NA layer error so maybe just HG_INVALID_ARG?